drivers/linstorvolumemanager.py

```diff
@@ -682,9 +682,28 @@ def destroy_volume(self, volume_uuid):
         Destroy a volume.
         :param str volume_uuid: The volume uuid to destroy.
         """
+        def get_volume_info(volume_uuid: str) -> Dict:
+            for volume in self.get_resources_info().values():
+                if volume["uuid"] == volume_uuid:
+                    return volume
+            raise LinstorVolumeManagerError(
+                f"Could not find info about volume `{volume_uuid}`",
+                LinstorVolumeManagerError.ERR_VOLUME_DESTROY
+            )
+
+        def is_volume_in_use(volume_info: Dict) -> bool:
+            for node in volume_info["nodes"].values():
+                if node["in-use"]:
+                    return True
+            return False
 
         self._ensure_volume_exists(volume_uuid)
         self.ensure_volume_is_not_locked(volume_uuid)
+        if is_volume_in_use(get_volume_info(volume_uuid)):
+            raise LinstorVolumeManagerError(
+                f"Could not destroy volume `{volume_uuid}` as it is currently in use",
+                LinstorVolumeManagerError.ERR_VOLUME_DESTROY
+            )
```

Comment:
Additionally, having single-use nested functions introduces unnecessary noise IMHO.

Don’t take this suggestion at face value, it's mostly meant to illustrate the idea.

```suggestion
        self._ensure_volume_exists(volume_uuid)
        self.ensure_volume_is_not_locked(volume_uuid)
        volume_info = next(
            (volume for volume in self.get_resources_info().values()
            if volume["uuid"] == volume_uuid), None
        )
        if not volume_info:
            raise LinstorVolumeManagerError(
                f"Could not find info about volume `{volume_uuid}`",
                LinstorVolumeManagerError.ERR_VOLUME_DESTROY
            )
        if any(node["in-use"] for node in volume_info["nodes"].values()):
            raise LinstorVolumeManagerError(
                f"Could not destroy volume `{volume_uuid}` as it is currently in use",
                LinstorVolumeManagerError.ERR_VOLUME_DESTROY
            )
```

Reply:
I made an in-between solution